### ISSUE TITLE
Server Crash Log: Using portalwand

### ISSUE THREAD

**Issue by pauldpickell** [ubc-minetest-classroom/Minetest_Classroom#17]
Partial output from today's crash:
```2022-05-24 20:46:04: ACTION[Server]: olson places node mc_realmportals:testRealmstone at (-18982,10,-19001)
2022-05-24 20:46:05: ACTION[Server]: olson places node mc_realmportals:testRealmstone at (-18982,11,-19001)
2022-05-24 20:46:06: ACTION[Server]: olson places node mc_realmportals:testRealmstone at (-18982,9,-18998)
2022-05-24 20:46:06: ACTION[Server]: olson places node mc_realmportals:testRealmstone at (-18982,10,-18998)
2022-05-24 20:46:06: ACTION[Server]: olson places node mc_realmportals:testRealmstone at (-18982,11,-18998)
2022-05-24 20:46:07: ACTION[Server]: olson places node mc_realmportals:testRealmstone at (-18982,12,-18998)
2022-05-24 20:46:08: ACTION[Server]: olson places node mc_realmportals:testRealmstone at (-18982,12,-19001)
2022-05-24 20:46:08: ACTION[Server]: olson places node mc_realmportals:testRealmstone at (-18982,12,-19000)
2022-05-24 20:46:09: ACTION[Server]: olson places node mc_realmportals:testRealmstone at (-18982,12,-18999)
2022-05-24 20:46:22: ACTION[Server]: olson invoked /give, stackstring="mc_realmportals:lukieRealmstone 64"
2022-05-24 20:46:24: ACTION[Server]: olson places node mc_realmportals:lukieRealmstone at (-18982,8,-18996)
2022-05-24 20:46:25: ACTION[Server]: olson places node mc_realmportals:lukieRealmstone at (-18982,8,-18995)
2022-05-24 20:46:25: ACTION[Server]: olson places node mc_realmportals:lukieRealmstone at (-18982,8,-18994)
2022-05-24 20:46:26: ACTION[Server]: olson places node mc_realmportals:lukieRealmstone at (-18982,8,-18993)
2022-05-24 20:46:27: ACTION[Server]: olson places node mc_realmportals:lukieRealmstone at (-18982,9,-18996)
2022-05-24 20:46:27: ACTION[Server]: olson places node mc_realmportals:lukieRealmstone at (-18982,10,-18996)
2022-05-24 20:46:28: ACTION[Server]: olson places node mc_realmportals:lukieRealmstone at (-18982,11,-18996)
2022-05-24 20:46:28: ACTION[Server]: olson places node mc_realmportals:lukieRealmstone at (-18982,12,-18996)
2022-05-24 20:46:29: ACTION[Server]: olson places node mc_realmportals:lukieRealmstone at (-18982,9,-18993)
2022-05-24 20:46:30: ACTION[Server]: olson places node mc_realmportals:lukieRealmstone at (-18982,10,-18993)
2022-05-24 20:46:30: ACTION[Server]: olson places node mc_realmportals:lukieRealmstone at (-18982,11,-18993)
2022-05-24 20:46:30: ACTION[Server]: olson places node mc_realmportals:lukieRealmstone at (-18982,12,-18993)
2022-05-24 20:46:32: ACTION[Server]: olson places node mc_realmportals:lukieRealmstone at (-18982,12,-18995)
2022-05-24 20:46:32: ACTION[Server]: olson places node mc_realmportals:lukieRealmstone at (-18982,12,-18994)
2022-05-24 20:46:38: ACTION[Server]: Prof_Pickell uses mc_student:notebook, pointing at [nothing]
2022-05-24 20:46:46: ACTION[Server]: olson invoked /give, stackstring="mc_realmportals:realm1024stone 64"
2022-05-24 20:46:49: ACTION[Server]: olson places node mc_realmportals:realm1024stone at (-18982,8,-19003)
2022-05-24 20:46:49: ACTION[Server]: olson places node mc_realmportals:realm1024stone at (-18982,8,-19004)
2022-05-24 20:46:50: ACTION[Server]: olson places node mc_realmportals:realm1024stone at (-18982,8,-19005)
2022-05-24 20:46:51: ACTION[Server]: olson places node mc_realmportals:realm1024stone at (-18982,9,-19003)
2022-05-24 20:46:51: ACTION[Server]: olson places node mc_realmportals:realm1024stone at (-18982,10,-19003)
2022-05-24 20:46:51: ACTION[Server]: olson places node mc_realmportals:realm1024stone at (-18982,11,-19003)
2022-05-24 20:46:53: ACTION[Server]: olson places node mc_realmportals:realm1024stone at (-18982,12,-19003)
2022-05-24 20:46:54: ACTION[Server]: olson places node mc_realmportals:realm1024stone at (-18982,8,-19006)
2022-05-24 20:46:55: ACTION[Server]: olson places node mc_realmportals:realm1024stone at (-18982,9,-19006)
2022-05-24 20:46:55: ACTION[Server]: olson places node mc_realmportals:realm1024stone at (-18982,10,-19006)
2022-05-24 20:46:56: ACTION[Server]: olson places node mc_realmportals:realm1024stone at (-18982,11,-19006)
2022-05-24 20:46:56: ACTION[Server]: olson places node mc_realmportals:realm1024stone at (-18982,12,-19006)
2022-05-24 20:46:57: ACTION[Server]: olson places node mc_realmportals:realm1024stone at (-18982,12,-19004)
2022-05-24 20:46:57: ACTION[Server]: olson places node mc_realmportals:realm1024stone at (-18982,12,-19005)
2022-05-24 20:47:04: ACTION[Server]: Daniel digs default:pine_needles at (-18900,17,-18958)
2022-05-24 20:47:05: ACTION[Server]: Daniel digs default:pine_needles at (-18901,17,-18957)
2022-05-24 20:47:05: ACTION[Server]: Daniel digs default:pine_needles at (-18900,17,-18957)
2022-05-24 20:47:07: ACTION[Server]: Daniel digs default:pine_needles at (-18900,18,-18957)
2022-05-24 20:47:08: ACTION[Server]: olson invoked /give, stackstring="portals:portalwand 64"
2022-05-24 20:47:10: ACTION[Main]: Server: Shutting down
2022-05-24 20:47:11: ERROR[Main]: ServerError: AsyncErr: Lua: Runtime error from mod 'portals' in callback item_OnPlace(): ...test_Classroom/mods/mc_realmportals/portalhelper.lua:110: attempt to call field 'CreateRalmByName' (a nil value)
2022-05-24 20:47:11: ERROR[Main]: stack traceback:
2022-05-24 20:47:11: ERROR[Main]:       ...test_Classroom/mods/mc_realmportals/portalhelper.lua:110: in function 'CreateGetRealm'
2022-05-24 20:47:11: ERROR[Main]:       ...test_Classroom/mods/mc_realmportals/portalhelper.lua:61: in function 'find_realm_anchorPos'
2022-05-24 20:47:11: ERROR[Main]:       ...ames/Minetest_Classroom/mods/portals/portals_api.lua:852: in function 'ignite_portal'
2022-05-24 20:47:11: ERROR[Main]:       ...ames/Minetest_Classroom/mods/portals/portals_api.lua:1399: in function <...ames/Minetest_Classroom/mods/portals/portals_api.lua:1396>```

**Comment by Lowie375** [comment]
From what I can see, a typo in [`mc_realmportals/portalhelper.lua:110`](https://github.com/ubc-minetest-classroom/Minetest_Classroom/blob/develop/mods/mc_realmportals/portalhelper.lua#L110) may have something to do with this: `CreateRalmByName` should be `CreateRealmByName`. This also applies to [line 116](https://github.com/ubc-minetest-classroom/Minetest_Classroom/blob/develop/mods/mc_realmportals/portalhelper.lua#L116) in the same file.

**Comment by lukasgolson** [comment]
Yup! This has all been fixed in a local branch. This was the issue.

On May 24, 2022, at 3:23 PM, Nanjou ***@***.***> wrote:

﻿

From what I can see, a typo in mc_realmportals/portalhelper.lua:110<https://github.com/ubc-minetest-classroom/Minetest_Classroom/blob/develop/mods/mc_realmportals/portalhelper.lua#L110> may have something to do with this: CreateRalmByName should be CreateRealmByName. This also applies to line 116<https://github.com/ubc-minetest-classroom/Minetest_Classroom/blob/develop/mods/mc_realmportals/portalhelper.lua#L116> in the same file.

**Comment by pauldpickell** [comment]
Good catch @Lowie375 
I fixed the typo on the server and was able to ignite two of the portals you built. `testRealmstone` and `realm1024stone` work just fine, but `lukieRealmstone` fails to ignite and crashes the server with the following output on the debug:

```-------------
  Separator
-------------

2022-05-24 22:46:22: [Main]: Using world specified by --worldname on the command line
2022-05-24 22:46:22: [Main]: Using game specified by --gameid on the command line
2022-05-24 22:46:22: WARNING[Main]: /!\ You are using the old mod storage files backend. This backend is deprecated and may be removed in a future release /!\
2022-05-24 22:46:22: WARNING[Main]: Switching to SQLite3 is advised, please read http://wiki.minetest.net/Database_backends.
2022-05-24 22:46:22: WARNING[Main]: Undeclared global variable "portalwand" accessed at ...in/../games/Minetest_Classroom/mods/portals/init.lua:52
2022-05-24 22:46:22: WARNING[Main]: tried registering a schematic with nil path for key: shack
2022-05-24 22:46:22: WARNING[Main]: Mod colorbrewer at /home/ubuntu/minetest/bin/../games/Minetest_Classroom/mods/colorbrewer:
2022-05-24 22:46:22: WARNING[Main]:     Mods not having a mod.conf file with the name is deprecated.
2022-05-24 22:46:22: ACTION[Main]: World at [/home/ubuntu/minetest/bin/../worlds/ubc_campus_mosaic]
2022-05-24 22:46:22: ACTION[Main]: Server for gameid="Minetest_Classroom" listening on 0.0.0.0:30000.
2022-05-24 22:46:36: ACTION[Server]: Prof_Pickell [128.189.115.130] joins game. List of players: Prof_Pickell
2022-05-24 22:46:39: ACTION[Server]: Prof_Pickell uses mc_teacher:controller, pointing at [nothing]
2022-05-24 22:46:52: ACTION[Server]: Prof_Pickell invoked /give, stackstring="portals:portalwand"
2022-05-24 22:46:58: ACTION[Server]: Prof_Pickell invoked /give, stackstring="portals:portalwand"
2022-05-24 22:47:52: ACTION[Server]: Prof_Pickell invoked /give, stackstring="portals:portalwand 2"
2022-05-24 22:47:53: ERROR[Server]: NodeResolver: failed to resolve node name 'stairs:slab_stone'.
2022-05-24 22:47:53: ERROR[Server]: NodeResolver: failed to resolve node name 'openstreetmaps:bench'.
2022-05-24 22:47:53: ERROR[Server]: NodeResolver: failed to resolve node name 'openstreetmaps:bicycle_parking'.
2022-05-24 22:47:56: ACTION[Server]: Prof_Pickell activates
2022-05-24 22:47:58: ACTION[Server]: Prof_Pickell invoked /give, stackstring="portals:portalwand 2"
2022-05-24 22:47:59: ACTION[Main]: Server: Shutting down
2022-05-24 22:48:00: ERROR[Main]: ServerError: AsyncErr: Lua: Runtime error from mod 'portals' in callback item_OnPlace(): ...es/Minetest_Classroom/mods/mc_worldmanager/realm.lua:269: attempt to index local 'config' (a nil value)
2022-05-24 22:48:00: ERROR[Main]: stack traceback:
2022-05-24 22:48:00: ERROR[Main]:       ...es/Minetest_Classroom/mods/mc_worldmanager/realm.lua:269: in function 'Load_Schematic'
2022-05-24 22:48:00: ERROR[Main]:       ...test_Classroom/mods/mc_realmportals/portalhelper.lua:128: in function <...test_Classroom/mods/mc_realmportals/portalhelper.lua:122>
2022-05-24 22:48:00: ERROR[Main]:       (tail call): ?
2022-05-24 22:48:00: ERROR[Main]:       ...test_Classroom/mods/mc_realmportals/portalhelper.lua:61: in function 'find_realm_anchorPos'
2022-05-24 22:48:00: ERROR[Main]:       ...ames/Minetest_Classroom/mods/portals/portals_api.lua:852: in function 'ignite_portal'
2022-05-24 22:48:00: ERROR[Main]:       ...ames/Minetest_Classroom/mods/portals/portals_api.lua:1399: in function <...ames/Minetest_Classroom/mods/portals/portals_api.lua:1396>
```

Seems the schematic is missing for the shack?

`2022-05-24 22:46:22: WARNING[Main]: tried registering a schematic with nil path for key: shack`

**Comment by lukasgolson** [comment]
Fixed in #20
